### PR TITLE
Make minlenratio effective

### PR DESCRIPTION
**Pull request by jctian98**
## What?
This is to respond to issues https://github.com/espnet/espnet/issues/5573 and https://github.com/espnet/espnet/issues/5580 to make the decoding hyper-parameter `minlenratio` effective in beam search

Previously the `minlenratio` and the variable `minlen` were calculated but not applied, which leads to the issue as described below:
Some models intend to predict the special symbol `<eos>` at the first several decoding steps. As long as the invalid `<eos>` remains in the top-k list, these invalid hypotheses are added to the ended hypothesis list. Since the invalid hypotheses are short, the accumulated posterior is also very high compared to other valid hypotheses with much longer text hypotheses. The invalid hypotheses are then selected in the finalization stage, most of which are empty strings. As a consequence, the overall decoding results would be empty or very short. The issue is more likely to be observed when beam size is large as the `<eos>` token is more likely to stay in the top-k list.

To overcome this issue, simply make the `minlenratio` hyperparameter effective by adding an `if` condition in the `post_process` function. Then pass a small heuristic value to it (e.g., 0.1). This will force all ended hypotheses to have a length of at least 10% of the input lengths.  Although it's heuristic, it works well generally.

This PR also adds a length normalization feature for the beam search object. If activated, the best hypothesis is selected by the best token-level score rather than the accumulated score.

### CODE REVIEW

**Comment by sw005320** [comment]
LGTM!
Once you fix the CI error, I'll merge it.

File: espnet/nets/batch_beam_search.py
Comment (sw005320) [inline espnet/nets/batch_beam_search.py:1]: Why it does not include the normalization?

**Comment by sw005320** [comment]
@siddhu001, can you check the case of negative `minlenratio`?
Would it be an expected behavior?

File: espnet/nets/batch_beam_search.py
Comment (jctian98) [reply espnet/nets/batch_beam_search.py:1]: I think the `batch_beam_search` object inherits the sort function from the `beam_search` object so. So we don't need to handle this in the `batch_beam_search` object.

**Comment by sw005320** [comment]
After @siddhu001's confirmation, I'll merge this PR.

**Comment by jctian98** [comment]
> @siddhu001, can you check the case of negative `minlenratio`? Would it be an expected behavior?

It's good to test with the attention-only model, as the ctc can greatly calibrate the length estimation so minlenratio will not be tested properly. 
Please let me know if there is any issue. Thanks @siddhu001

**Comment by siddhu001** [comment]
> @siddhu001, can you check the case of negative `minlenratio`? Would it be an expected behavior?

Hi @sw005320,

Yes I believe the case of negative minlenratio is the expected behaviour. We follow a similar approach for negative maxlenratio (i.e. maxlenratio=-1*maxlenratio when maxlenratio<0)

@jctian98 thanks a lot for your efforts on this PR. This is very useful!
